src/main/kotlin/it/pagopa/wallet/services/WalletPaymentCDCEventDispatcherService.kt

```diff
@@ -1,44 +1,56 @@
 package it.pagopa.wallet.services
 
-import com.azure.core.http.rest.Response
-import com.azure.storage.queue.models.SendMessageResult
 import it.pagopa.wallet.client.WalletQueueClient
 import it.pagopa.wallet.common.tracing.TracingUtils
+import it.pagopa.wallet.config.RetrySendPolicyConfig
 import it.pagopa.wallet.config.properties.CdcQueueConfig
 import java.time.Duration
 import org.bson.BsonDocument
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Component
 import reactor.core.publisher.Mono
+import reactor.util.retry.Retry
 
 @Component
 class WalletPaymentCDCEventDispatcherService(
     private val walletQueueClient: WalletQueueClient,
     private val tracingUtils: TracingUtils,
     private val cdcQueueConfig: CdcQueueConfig,
+    private val retrySendPolicyConfig: RetrySendPolicyConfig
 ) {
 
     private val WALLET_CDC_EVENT_HANDLER_SPAN_NAME = "cdcWalletEvent"
     private val logger: Logger = LoggerFactory.getLogger(javaClass)
 
-    private val walletExpireTimeout = Duration.ofSeconds(cdcQueueConfig.timeoutWalletExpired)
+    private val walletCdcTimeout = Duration.ofSeconds(cdcQueueConfig.timeoutWalletCdc)
 
-    fun dispatchEvent(event: BsonDocument): Mono<BsonDocument> =
+    fun dispatchEvent(event: BsonDocument?): Mono<BsonDocument> =
         if (event != null) {
-            onWalletEvent(event).map { event }
+            Mono.defer {
+                    tracingUtils.traceMonoQueue(WALLET_CDC_EVENT_HANDLER_SPAN_NAME) { tracingInfo ->
+                        walletQueueClient.sendWalletEvent(
+                            event = event,
+                            delay = walletCdcTimeout,
+                            tracingInfo = tracingInfo,
+                        )
+                    }
+                }
+                .retryWhen(
+                    Retry.fixedDelay(
+                            retrySendPolicyConfig.maxAttempts,
+                            Duration.ofMillis(retrySendPolicyConfig.intervalInMs)
+                        )
+                        .filter { t -> t is Exception }
+                        .doBeforeRetry { signal ->
+                            logger.info(
+                                "Retrying writing event on CDC queue due to: ${signal.failure().message}"
+                            )
+                        }
+                )
+                .doOnError { e -> logger.error("Failed to send event after retries", e) }
```

Comment:
in this log is it possible to log the body of the message? which could potentially contain data not to be shown in the logs?

Reply:
sure. i edited this to only print the message of the exception
here 65437136d97d5e8b004b7d34afd4ed37b94fb5e6